examples/prom/maxwell_global_rom.cpp

```diff
@@ -0,0 +1,538 @@
+/******************************************************************************
+ *
+ * Copyright (c) 2013-2023, Lawrence Livermore National Security, LLC
+ * and other libROM project developers. See the top-level COPYRIGHT
+ * file for details.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR MIT)
+ *
+ *****************************************************************************/
+
+//               libROM MFEM Example: parametric ROM for Maxwell equation (adapted from ex3p.cpp)
+//
+// Compile with: make maxwell_global_rom
+//
+// Description:  This example code uses MFEM and librom to define a projection-based
+//               reduced-order model for a simple electromagnetic diffusion
+//               problem corresponding to the second order definite Maxwell
+//               equation: curl curl E + E = f. The boundary condition is given by
+//               E x n = <given tangential field>. Here, we use a given exact
+//               solution E and compute the corresponding r.h.s. f. Each component
+//               of the vector field E is assumed to be a sin wave, with frequency
+//               controlled by the parameter kappa.
+//
+//               The example highlights three distinct ROM processes - the
+//               offline, merge, and online phases. The offline phase runs the full
+//               order model and stores the snapshot data in an HDF file. You
+//               can run as many offline phases as you wish to sample the
+//               parameter space. The merge phase reads all the snapshot files,
+//               builds a global reduced basis, and stores the basis in an HDF
+//               file. The online phase reads the basis, builds the ROM
+//               operator, solves the reduced order system, and lifts the
+//               solution to the full order space.
+//
+// For ROM (global rom):
+// Offline phase: maxwell_global_rom -offline -f 1.0 -id 0
+//                maxwell_global_rom -offline -f 1.1 -id 1
+//                maxwell_global_rom -offline -f 1.2 -id 2
+//
+// Merge phase:   maxwell_global_rom -merge -ns 3
+//
+// FOM solution:  maxwell_global_rom -fom -f 1.15
+//
+// Online phase:  maxwell_global_rom -online -f 1.15
+//
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+#include "linalg/Vector.h"
+#include "linalg/BasisGenerator.h"
+#include "linalg/BasisReader.h"
+#include "mfem/Utilities.hpp"
+#include <cmath>
+#include <set>
+
+using namespace std;
+using namespace mfem;
+
+// Exact solution, E, and r.h.s., f. See below for implementation.
+void E_exact(const Vector &, Vector &);
+void f_exact(const Vector &, Vector &);
+double freq = 1.0, kappa;
+int dim;
+
+int main(int argc, char *argv[])
+{
+   // 1. Initialize MPI and HYPRE
+   Mpi::Init(argc, argv);
+   int num_procs = Mpi::WorldSize();
+   int myid = Mpi::WorldRank();
+   Hypre::Init();
+
+   // 2. Parse command-line options.
+//   const char *mesh_file = "../data/beam-tet.mesh";
```

Comment:
You can remove this line

Reply:
I ran the stylization and removed line 71